python/lsst/ctrl/mpexec/cli/opt/options.py

```diff
@@ -440,6 +445,46 @@
     help=unwrap("""Names of input dataset types that should not be mocked."""),
 )
 
+
+def parse_mock_failure(
+    ctx: click.Context, param: click.Option, value: Iterable[str] | None
+) -> Mapping[str, tuple[str, type[Exception] | None]]:
+    """Parse the --mock-failure option values into the mapping accepted by
+    `lsst.pipe.base.tests.mocks.mock_task_defs`.
+    """
+    result: dict[str, tuple[str, type[Exception] | None]] = {}
+    if value is None:
+        return result
+    for entry in value:
+        try:
+            task_label, error_type_name, where = entry.split(":")
```

Comment:
One more comment - what if `where` also contains colon (e.g. for slicing)?

Reply:
Good catch.  I've updated this to
```
task_label, error_type_name, *where_terms = entry.split(":")
where = ":".join(where_terms)
```
to guard against that.

Reply:
`entry.split(":", 2)` is probably faster 🙂

Reply:
Neat, I didn't know about that argument.